wasmplugin/config.go

```diff
@@ -40,5 +57,39 @@ func parsePluginConfiguration(data []byte) (pluginConfiguration, error) {
 		return true
 	})
 
+	defaultDirective := jsonData.Get("default_directive")
+	if defaultDirective.Exists() {
+		defaultDirectiveName := defaultDirective.String()
+		if _, ok := config.directivesMap[defaultDirectiveName]; !ok {
+			return config, fmt.Errorf("directive map not found for default directive: %q", defaultDirectiveName)
+		}
+
+		config.defaultDirective = defaultDirectiveName
+	}
+
+	config.perAuthorityDirectives = make(map[string]string)
+	jsonData.Get("per_authority_directives").ForEach(func(key, value gjson.Result) bool {
+		config.perAuthorityDirectives[key.String()] = value.String()
+		return true
+	})
+
+	for authority, directiveName := range config.perAuthorityDirectives {
+		if _, ok := config.directivesMap[directiveName]; !ok {
+			return config, fmt.Errorf("directive map not found for authority %s: %q", authority, directiveName)
+		}
+	}
+
+	rules := jsonData.Get("rules")
```

Comment:
nit: only get the rules from the json if the directives map is empty. there is no need to get it otherwise.

Reply:
Thanks! Looks good!